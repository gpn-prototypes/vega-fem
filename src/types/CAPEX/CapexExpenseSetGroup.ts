import Article, { ArticleValues } from '../Article';

export default interface CapexExpenseSetGroup {
  id?: string | number;
  name?: string;
  caption?: string;
  valueTotal?: number;
  capexExpenseList?: Article[];
  totalValueByYear?: ArticleValues[];
}
