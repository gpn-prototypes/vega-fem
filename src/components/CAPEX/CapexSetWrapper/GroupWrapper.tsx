import React, { useCallback, useState } from 'react';
import { IconArrowDown, Text, useModal } from '@gpn-prototypes/vega-ui';

import { GroupPlaceholder } from '../../Macroparameters/MacroparameterSetWrapper/GroupPlaceholder/GroupPlaceholder';
import { cnGroupWrapper } from '../../Macroparameters/MacroparameterSetWrapper/GroupWrapper/cn-group-wrapper';
import { Collapsed } from '../../Macroparameters/MacroparameterSetWrapper/GroupWrapper/GroupWrapper';
import { ArticleWrapper } from '../../Shared/Article/ArticleWrapper';
import { AddArticleModal } from '../../Shared/GroupOptionsDropdown/AddArticleModal/AddArticleModal';
import { GroupOptionsDropdown } from '../../Shared/GroupOptionsDropdown/GroupOptionsDropdown';

import '@/styles/BlockWrapper/BlockWrapper.css';
import '../../Macroparameters/MacroparameterSetWrapper/GroupWrapper/GroupWrapper.css';

import keyGen from '@/helpers/keyGenerator';
import Article from '@/types/Article';
import CapexExpenseSetGroup from '@/types/CAPEX/CapexExpenseSetGroup';
import MacroparameterSetGroup from '@/types/Macroparameters/MacroparameterSetGroup';

export interface CapexSetWrapperGroupProps {
  group: CapexExpenseSetGroup;
  requestAddCapex: (capex: Article, group: CapexExpenseSetGroup) => void;
  updateCapexValue: (capex: Article, group: CapexExpenseSetGroup) => void;
  deleteCapexValue: (capex: Article, group: CapexExpenseSetGroup) => void;
  requestChangeCapexGroup: (group: CapexExpenseSetGroup) => void;
  requestDeleteCapexGroup: (group: CapexExpenseSetGroup) => void;
  onArticleFocusCallback?: (article: Article, group: MacroparameterSetGroup) => void;
  highlightArticleClear?: () => void;
  isCollapsed?: Collapsed;
  isCollapsedCallback?: (collapsed: Collapsed) => void;
}

export const GroupWrapper = ({
  group,
  requestAddCapex,
  updateCapexValue,
  deleteCapexValue,
  requestChangeCapexGroup,
  requestDeleteCapexGroup,
  onArticleFocusCallback,
  highlightArticleClear,
  isCollapsed,
  isCollapsedCallback,
}: CapexSetWrapperGroupProps) => {
  const [capexes] = useState((group?.capexExpenseList ?? []) as Article[]);

  const [isCollapsedState, setIsCollapsedState] = useState(isCollapsed?.collapsed ?? true);

  const { isOpen, close, open } = useModal();

  const openAddCapexModal = (): void => {
    setIsCollapsedState(false);
    open();
  };
  const addCapexToGroup = (capex: Article): void => requestAddCapex(capex, group);

  const updateCapexValueWithGroup = (capex: Article): void => updateCapexValue(capex, group);

  const deleteCapexValueWithGroup = (capex: Article): void => deleteCapexValue(capex, group);

  const articleFocusHandler = useCallback(
    (article: Article) => {
      if (onArticleFocusCallback) {
        onArticleFocusCallback(article, group);
      }
    },
    [onArticleFocusCallback, group],
  );

  const onToggleCollapse = () => {
    setIsCollapsedState((prev) => !prev);
    if (isCollapsedCallback) {
      isCollapsedCallback({ id: isCollapsed?.id, collapsed: !isCollapsed?.collapsed } as Collapsed);
    }
  };

  return (
    <div className={cnGroupWrapper()}>
      <div className={cnGroupWrapper('header')}>
        <div
          className={cnGroupWrapper('header-name', { collapse: isCollapsedState })}
          onClick={onToggleCollapse}
          role="presentation"
        >
          <IconArrowDown size="xs" />
          <Text as="span" size="s">
            {group?.caption}
          </Text>
        </div>
        <div className={cnGroupWrapper('header-actions')}>
          <GroupOptionsDropdown<CapexExpenseSetGroup>
            group={group}
            requestAddArticle={requestAddCapex}
            requestChangeGroup={requestChangeCapexGroup}
            requestDeleteGroup={requestDeleteCapexGroup}
          />
        </div>
      </div>
      <div className={cnGroupWrapper('body', { hidden: isCollapsedState })}>
        {capexes?.length === 0 && (
          <GroupPlaceholder text="Пока не добавлена ни одна статья" callback={openAddCapexModal} />
        )}
        {capexes?.length > 0 &&
          capexes.map((article: Article, index: any) => (
            <ArticleWrapper
              key={keyGen(index)}
              article={article}
              fullWidth
              onFocusCallback={articleFocusHandler}
              highlightArticleClear={highlightArticleClear}
              updateArticleValueCallback={updateCapexValueWithGroup}
              updateArticleCallback={updateCapexValueWithGroup}
              deleteArticleCallback={deleteCapexValueWithGroup}
            />
          ))}
      </div>
      <AddArticleModal isOpen={isOpen} close={close} callback={addCapexToGroup} />
    </div>
  );
};
