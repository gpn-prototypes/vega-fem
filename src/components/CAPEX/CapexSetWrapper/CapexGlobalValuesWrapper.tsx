import React, { useCallback, useState } from 'react';
import { Form, TextField } from '@gpn-prototypes/vega-ui';

import CapexSetGlobalValue from '../../../../types/CAPEX/CapexSetGlobalValue';
import { prepareStringForBack, spreadValue } from '../../../helpers/spreadValue';
import { cnVegaFormCustom } from '../../../styles/VegaFormCustom/cn-vega-form-custom';

import '../../../styles/BlockWrapper/BlockWrapper.css';
import '../../Macroparameters/MacroparameterSetWrapper/GroupWrapper/GroupWrapper.css';

interface CapexWrapperProps {
  globalValue: CapexSetGlobalValue;
  updateCapexGlobalValue: (reserveValue: CapexSetGlobalValue) => void;
}

export const CapexGlobalValuesWrapper = ({
  globalValue,
  updateCapexGlobalValue,
}: CapexWrapperProps) => {
  const [value, setValue] = useState(globalValue?.value);
  const [spreaded, setSpreaded] = useState<boolean>(true);

  const editValues = (e: any): void => {
    if (spreaded) setSpreaded(false);
    setValue(e.e.target.value);
  };
  const requestSetGlobalValue = useCallback(() => {
    updateCapexGlobalValue({
      id: globalValue.id,
      value: typeof value === 'string' ? prepareStringForBack(value) : value,
    } as CapexSetGlobalValue);
    setSpreaded(true);
  }, [updateCapexGlobalValue, value, globalValue]);

  const loseFocus = (e: any) => {
    // TODO: change any
    if (e.key === 'Enter') {
      (e.target as HTMLElement).blur();
    }
  };

  return (
    <Form.Row gap="m" space="none" className={cnVegaFormCustom('form-row')}>
      <Form.Field>
        <Form.Label htmlFor={`capexSet${globalValue.name}`}>{globalValue.caption}</Form.Label>
        <TextField
          id={`capexSet${globalValue.name}`}
          size="s"
          width="full"
          value={value && spreaded ? spreadValue(value) : value?.toString()}
          rightSide={globalValue.unit ?? ''}
          onBlur={() => requestSetGlobalValue()}
          onChange={(e) => editValues(e)}
          onKeyDown={(e) => loseFocus(e)}
        />
      </Form.Field>
    </Form.Row>
  );
};
