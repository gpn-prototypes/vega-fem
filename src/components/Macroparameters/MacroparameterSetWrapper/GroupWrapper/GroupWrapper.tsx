import React, { useCallback, useState } from 'react';
import { IconArrowDown } from '@gpn-design/uikit/IconArrowDown';
import { Text, useModal } from '@gpn-prototypes/vega-ui';

import Article from '../../../../../types/Article';
import MacroparameterSetGroup from '../../../../../types/Macroparameters/MacroparameterSetGroup';
import keyGen from '../../../../helpers/keyGenerator';
import { ArticleWrapper } from '../../../Shared/Article/ArticleWrapper';
import { AddArticleModal } from '../../../Shared/GroupOptionsDropdown/AddArticleModal/AddArticleModal';
import { GroupOptionsDropdown } from '../../../Shared/GroupOptionsDropdown/GroupOptionsDropdown';
import { GroupPlaceholder } from '../GroupPlaceholder/GroupPlaceholder';

import { cnGroupWrapper } from './cn-group-wrapper';

import '../../../../styles/BlockWrapper/BlockWrapper.css';
import './GroupWrapper.css';

export interface Collapsed {
  id: string | number;
  collapsed: boolean;
}

interface MacroparameterSetWrapperGroupProps {
  group: MacroparameterSetGroup;
  removeGroup: (group: MacroparameterSetGroup) => void;
  requestAddMacroparameter: (macroparameter: Article, group: MacroparameterSetGroup) => void;
  updateMacroparameterValue: (macroparameter: Article, group: MacroparameterSetGroup) => void;
  deleteMacroparameterValue: (macroparameter: Article, group: MacroparameterSetGroup) => void;
  requestChangeMacroparameterGroup: (group: MacroparameterSetGroup) => void;
  requestDeleteMacroparameterGroup: (group: MacroparameterSetGroup) => void;
  onArticleFocusCallback?: (article: Article, group: MacroparameterSetGroup) => void;
  highlightArticleClear?: () => void;
  isCollapsed?: Collapsed;
  isCollapsedCallback?: (collapsed: Collapsed) => void;
}

export const GroupWrapper = ({
  group,
  requestAddMacroparameter,
  updateMacroparameterValue,
  deleteMacroparameterValue,
  requestChangeMacroparameterGroup,
  requestDeleteMacroparameterGroup,
  onArticleFocusCallback,
  highlightArticleClear,
  isCollapsed,
  isCollapsedCallback,
}: MacroparameterSetWrapperGroupProps) => {
  const [articles] = useState(group.macroparameterList as Article[]);

  const [isCollapsedState, setIsCollapsedState] = useState(isCollapsed?.collapsed ?? true);

  const { isOpen, close, open } = useModal();

  const openAddMacroparameterModal = (): void => {
    setIsCollapsedState(false);
    open();
  };

  const addMacroparameterToGroup = (macroparameter: Article): void =>
    requestAddMacroparameter(macroparameter, group);

  const updateMacroparameterValueWithGroup = (macroparameter: Article): void =>
    updateMacroparameterValue(macroparameter, group);

  const deleteMacroparameterValueWithGroup = (macroparameter: Article): void =>
    deleteMacroparameterValue(macroparameter, group);

  const articleFocusHandler = useCallback(
    (article: Article) => {
      if (onArticleFocusCallback) {
        onArticleFocusCallback(article, group);
      }
    },
    [onArticleFocusCallback, group],
  );

  const onToggleCollapse = () => {
    setIsCollapsedState((prev) => !prev);
    if (isCollapsedCallback) {
      isCollapsedCallback({ id: isCollapsed?.id, collapsed: !isCollapsed?.collapsed } as Collapsed);
    }
  };

  return (
    <div className={cnGroupWrapper()}>
      <div className={cnGroupWrapper('header')}>
        <div
          className={cnGroupWrapper('header-name', { collapse: isCollapsedState })}
          onClick={onToggleCollapse}
          role="presentation"
        >
          <IconArrowDown size="xs" />
          <Text as="span" size="s">
            {group.caption}
          </Text>
        </div>
        <div className={cnGroupWrapper('header-actions')}>
          <GroupOptionsDropdown<MacroparameterSetGroup>
            group={group}
            requestAddArticle={addMacroparameterToGroup}
            requestChangeGroup={requestChangeMacroparameterGroup}
            requestDeleteGroup={requestDeleteMacroparameterGroup}
          />
        </div>
      </div>
      <div className={cnGroupWrapper('body', { hidden: isCollapsedState })}>
        {articles?.length === 0 && (
          <GroupPlaceholder
            text="Пока не добавлена ни одна статья"
            callback={openAddMacroparameterModal}
          />
        )}
        {articles?.length > 0 &&
          articles.map((macroparameter, index) => (
            <ArticleWrapper
              key={keyGen(index)}
              article={macroparameter}
              updateArticleValueCallback={updateMacroparameterValueWithGroup}
              updateArticleCallback={updateMacroparameterValueWithGroup}
              deleteArticleCallback={deleteMacroparameterValueWithGroup}
              onFocusCallback={articleFocusHandler}
              highlightArticleClear={highlightArticleClear}
            />
          ))}
      </div>
      <AddArticleModal isOpen={isOpen} close={close} callback={addMacroparameterToGroup} />
    </div>
  );
};
