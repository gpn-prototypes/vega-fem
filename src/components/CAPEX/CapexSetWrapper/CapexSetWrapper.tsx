import React, { useEffect, useState } from 'react';
import { Button, Form, IconAdd, IconSelect, Text, TextField } from '@gpn-prototypes/vega-ui';

import Article from '../../../../types/Article';
import CapexExpenseSetGroup from '../../../../types/CAPEX/CapexExpenseSetGroup';
import CapexSet from '../../../../types/CAPEX/CapexSet';
import CapexSetGlobalValue from '../../../../types/CAPEX/CapexSetGlobalValue';
import MacroparameterSetGroup from '../../../../types/Macroparameters/MacroparameterSetGroup';
import { CapexTableContainer } from '../../../containers/CAPEX/CapexTableContainer';
import keyGen from '../../../helpers/keyGenerator';
import { cnBlockWrapper } from '../../../styles/BlockWrapper/cn-block-wrapper';
import { cnVegaFormCustom } from '../../../styles/VegaFormCustom/cn-vega-form-custom';
import { Collapsed } from '../../Macroparameters/MacroparameterSetWrapper/GroupWrapper/GroupWrapper';
import { validateName } from '../../Shared/ErrorMessage/ValidateArticle';
import { Validation } from '../../Shared/ErrorMessage/Validation';

import { CapexGlobalValuesWrapper } from './CapexGlobalValuesWrapper';
import { GroupWrapper } from './GroupWrapper';

import '../../../styles/BlockWrapper/BlockWrapper.css';
import '../../../styles/VegaFormCustom/VegaFormCustom.css';

export interface CapexSetWrapperProps {
  capexSet: CapexSet;
  addCapexSetGroup: (capexSetGroup: CapexExpenseSetGroup) => void;
  addCapex: (capex: Article, group: CapexExpenseSetGroup) => void;
  updateCapexGlobalValue: (reserveValue: CapexSetGlobalValue) => void;
  updateCapexValue: (capex: Article, group: CapexExpenseSetGroup) => void;
  deleteCapexValue: (capex: Article, group: CapexExpenseSetGroup) => void;
  requestChangeCapexGroup: (group: CapexExpenseSetGroup) => void;
  requestDeleteCapexGroup: (group: CapexExpenseSetGroup) => void;
  highlightArticle: (article: Article, group: MacroparameterSetGroup) => void;
  highlightArticleClear: () => void;
}

export const CapexSetWrapper = ({
  capexSet,
  updateCapexGlobalValue,
  addCapexSetGroup,
  addCapex,
  updateCapexValue,
  deleteCapexValue,
  requestChangeCapexGroup,
  requestDeleteCapexGroup,
  highlightArticle,
  highlightArticleClear,
}: CapexSetWrapperProps) => {
  const [isAddingGroup, setIsAddingGroup] = useState(false);
  const [newGroupName, setNewGroupName] = useState<string | undefined>('');
  const [groups, setGroups] = useState(
    (capexSet?.capexExpenseGroupList ?? []) as CapexExpenseSetGroup[],
  );
  const [globalValues, setGlobalValues] = useState(
    (capexSet?.capexGlobalValueList ?? []) as CapexSetGlobalValue[],
  );

  /* collapse/expand groups state */
  const [groupsCollapsed, setGroupsCollapsed] = useState([] as Collapsed[]);

  useEffect(() => {
    setGroups(capexSet?.capexExpenseGroupList ?? []);
    setGlobalValues(capexSet?.capexGlobalValueList ?? []);
    setGroupsCollapsed((prev) =>
      (capexSet?.capexExpenseGroupList ?? [])?.map(
        (group) =>
          ({
            id: group.id,
            collapsed: prev.filter((collapsed) => collapsed.id === group.id)[0]?.collapsed ?? true,
          } as Collapsed),
      ),
    );
  }, [capexSet]);

  const toggleCapexSetGroup = (event: React.MouseEvent) => {
    event.preventDefault();
    setIsAddingGroup(!isAddingGroup);
    setNewGroupName('');
  };

  const requestGroupAdd = (groupName: string) => {
    addCapexSetGroup({
      caption: groupName,
    } as CapexExpenseSetGroup);
  };

  const addGroup = (event: any, groupName: string): void => {
    toggleCapexSetGroup(event);
    requestGroupAdd(groupName);
  };

  const isCollapsedCallback = (collapsed: Collapsed) => {
    setGroupsCollapsed((prev) =>
      prev?.map((prevCollapsedItem) => {
        if (prevCollapsedItem.id === collapsed.id) {
          return { id: prevCollapsedItem.id, collapsed: collapsed.collapsed } as Collapsed;
        }
        return prevCollapsedItem;
      }),
    );
  };

  return (
    <div className={cnBlockWrapper()}>
      <div className={cnBlockWrapper('title-wrapper')}>
        <div className={cnBlockWrapper('title')}>
          <Text as="span" size="xs" className={cnBlockWrapper('title-text')}>
            CAPEX
          </Text>
          <IconSelect size="xs" className={cnBlockWrapper('title-icon')} />
        </div>
      </div>
      <div className={cnBlockWrapper('content')}>
        {capexSet && !(Object.keys(capexSet).length === 0) ? (
          <>
            <Form
              className={`${cnVegaFormCustom()} ${cnBlockWrapper('content-column')}`}
              onSubmit={(e: React.FormEvent) => {
                e.preventDefault();
              }}
            >
              <Form.Row gap="m" space="none" className={cnVegaFormCustom('content-body')}>
                {(globalValues ?? []).length > 0 &&
                  globalValues.map(
                    (globalValue: CapexSetGlobalValue, index: number) =>
                      index < 2 && (
                        <CapexGlobalValuesWrapper
                          key={keyGen(index)}
                          globalValue={globalValue}
                          updateCapexGlobalValue={updateCapexGlobalValue}
                        />
                      ),
                  )}
                <Form.Row
                  col="1"
                  gap="none"
                  space="none"
                  className={cnVegaFormCustom('groups-row')}
                >
                  {(groups ?? []).length > 0 &&
                    groups.map((group, index) => (
                      <GroupWrapper
                        key={keyGen(index)}
                        group={group}
                        groupList={groups}
                        requestAddCapex={addCapex}
                        updateCapexValue={updateCapexValue}
                        deleteCapexValue={deleteCapexValue}
                        requestChangeCapexGroup={requestChangeCapexGroup}
                        requestDeleteCapexGroup={requestDeleteCapexGroup}
                        onArticleFocusCallback={highlightArticle}
                        highlightArticleClear={highlightArticleClear}
                        isCollapsed={
                          groupsCollapsed.filter((collapsed) => collapsed.id === group.id)[0]
                        }
                        isCollapsedCallback={isCollapsedCallback}
                      />
                    ))}
                </Form.Row>
              </Form.Row>
              <Form.Row col="1" gap="none" space="none" className={cnVegaFormCustom('footer')}>
                {!isAddingGroup && (
                  <Button
                    type="button"
                    size="s"
                    label="Добавить группу затрат"
                    iconLeft={IconAdd}
                    view="ghost"
                    onClick={(e) => toggleCapexSetGroup(e)}
                  />
                )}
                {isAddingGroup && (
                  <Form.Field>
                    <Form.Label size="s">Название группы затрат</Form.Label>
                    <Form.Row col="1" gap="none" className={cnVegaFormCustom('footer-text-field')}>
                      <Validation
                        validationFunction={validateName}
                        linkedHook={setNewGroupName}
                        isClear
                        itemsList={groups}
                      >
                        <TextField
                          id="capexSetNewGroupName"
                          size="s"
                          width="full"
                          value={newGroupName}
                          maxLength={257}
                          placeholder="Введите название группы затрат"
                          type="text"
                          // onChange={(event: any) => setNewGroupName(event.e.target.value)}
                        />
                      </Validation>
                    </Form.Row>
                    <Form.Row className={cnVegaFormCustom('footer-action')}>
                      <Button
                        size="s"
                        label="Добавить группу"
                        view="ghost"
                        disabled={!newGroupName?.length}
                        onClick={(e) => addGroup(e, newGroupName ?? '')}
                      />
                      <Button label="Отмена" size="s" view="clear" onClick={toggleCapexSetGroup} />
                    </Form.Row>
                  </Form.Field>
                )}
              </Form.Row>
            </Form>
            <CapexTableContainer capexSet={capexSet} />
          </>
        ) : (
          <div />
          // <CapexSetPlaceholder text="Выберите один из макроэкономических сценариев" />
        )}
      </div>
    </div>
  );
};
