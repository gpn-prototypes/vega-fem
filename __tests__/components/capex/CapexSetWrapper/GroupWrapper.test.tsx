import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import { GroupWrapper } from '../../../../src/components/CAPEX/CapexSetWrapper/GroupWrapper';
import CapexExpenseSetGroup from '../../../../types/CAPEX/CapexExpenseSetGroup';
// import {render,screen,fireEvent} from "@testing-library/react";

let fakeGroup: CapexExpenseSetGroup;

beforeEach(() => {
  fakeGroup = {
    id: '1',
    name: 'oneTimePaymentGroup',
    caption: 'Первоначальный взнос',
    valueTotal: 1001000.0,
    capexExpenseList: [
      {
        id: '1',
        name: 'oneTimePaymentValue',
        caption: 'Значение разового платежа',
        valueTotal: 1000000.0,
      },
      {
        id: '2',
        name: 'secondTimePaymentValue',
        caption: 'Значение второго платежа',
        valueTotal: 1000,
      },
    ],
  };
});

describe('GroupWrapper', () => {
  test('рендерится без ошибок', () => {
    const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer
      .create(
        // <GroupWrapper group={fakeGroup} requestAddCapex={jest.fn()} updateCapexValue={jest.fn()} />,
        <GroupWrapper
          group={fakeGroup}
          deleteCapexValue={jest.fn()}
          requestAddCapex={jest.fn()}
          requestChangeCapexGroup={jest.fn()}
          requestDeleteCapexGroup={jest.fn()}
          updateCapexValue={jest.fn()}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('корректно рендерится при остутствии элементов в группе', () => {
    render(
      <GroupWrapper
        group={{ ...fakeGroup, capexExpenseList: [] }}
        deleteCapexValue={jest.fn()}
        requestAddCapex={jest.fn()}
        requestChangeCapexGroup={jest.fn()}
        requestDeleteCapexGroup={jest.fn()}
        updateCapexValue={jest.fn()}
      />,
    );
    expect(screen.getByText(/Пока не добавлена ни одна статья/i)).toBeInTheDocument();
  });
  /*  test('работает добавление статьи', () => {
    const fakeAdd = jest.fn();//TODO:fix integration test to

    render(<GroupWrapper group={fakeGroup} requestAddCapex={fakeAdd} updateCapexValue={() => {
    }}/>);

    const addArticleButton=screen.getByTitle(/добавить статью/i);
    fireEvent.click(addArticleButton);
    expect()
  }); */
});
