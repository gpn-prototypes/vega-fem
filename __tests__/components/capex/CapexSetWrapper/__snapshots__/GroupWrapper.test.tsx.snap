// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupWrapper рендерится без ошибок 1`] = `
<div
  className="GroupWrapper"
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
      onClick={[Function]}
      role="presentation"
    >
      <span
        className="Icon Icon_size_xs IconArrowDown"
      >
        <svg
          className="Icon-Svg"
          viewBox="0 0 12 12"
        >
          <path
            clipRule="evenodd"
            d="M6.005 8.341L2.03 4.366l.707-.707 3.268 3.268 3.268-3.268.707.707-3.975 3.975z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <span
        className="Text text text_size_s text_view_primary"
      >
        Первоначальный взнос
      </span>
    </div>
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <button
        className="Button Button_size_s Button_view_ghost Button_form_default Button_withIcon Button_onlyIcon MixFocus"
        onClick={[Function]}
        title="Опции"
        type="button"
      >
        <span
          className="Icon Icon_size_xs IconKebab Button-Icon"
        >
          <svg
            className="Icon-Svg"
            viewBox="0 0 16 16"
          >
            <path
              d="M8 4.5c.83 0 1.5-.67 1.5-1.5S8.83 1.5 8 1.5 6.5 2.17 6.5 3 7.17 4.5 8 4.5zm0 2c-.83 0-1.5.67-1.5 1.5S7.17 9.5 8 9.5 9.5 8.83 9.5 8 8.83 6.5 8 6.5zm0 5c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <div
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        <label
          className={
            Object {
              "has": [Function],
              "is": [Function],
              "mix": [Function],
              "split": [Function],
              "state": [Function],
              "toString": [Function],
            }
          }
          data-testid="label"
        >
          Значение разового платежа
        </label>
        <div
          className={
            Object {
              "has": [Function],
              "is": [Function],
              "mix": [Function],
              "split": [Function],
              "state": [Function],
              "toString": [Function],
            }
          }
        >
          <div
            className={
              Object {
                "has": [Function],
                "is": [Function],
                "mix": [Function],
                "split": [Function],
                "state": [Function],
                "toString": [Function],
              }
            }
          >
            <div
              className="TextField TextField_size_s TextField_view_default TextField_form_default TextField_width_full TextField_type_text"
              data-testid="input"
              onKeyDown={[Function]}
            >
              <input
                autoFocus={false}
                className="TextField-Input"
                id="article_oneTimePaymentValue_1"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Значение"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className={
              Object {
                "has": [Function],
                "is": [Function],
                "mix": [Function],
                "split": [Function],
                "state": [Function],
                "toString": [Function],
              }
            }
          >
            <button
              className="Button Button_size_s Button_view_ghost Button_form_default Button_withIcon Button_onlyIcon MixFocus"
              onClick={[Function]}
              title="Опции"
              type="button"
            >
              <span
                className="Icon Icon_size_xs IconKebab Button-Icon"
              >
                <svg
                  className="Icon-Svg"
                  viewBox="0 0 16 16"
                >
                  <path
                    d="M8 4.5c.83 0 1.5-.67 1.5-1.5S8.83 1.5 8 1.5 6.5 2.17 6.5 3 7.17 4.5 8 4.5zm0 2c-.83 0-1.5.67-1.5 1.5S7.17 9.5 8 9.5 9.5 8.83 9.5 8 8.83 6.5 8 6.5zm0 5c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <div
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        <label
          className={
            Object {
              "has": [Function],
              "is": [Function],
              "mix": [Function],
              "split": [Function],
              "state": [Function],
              "toString": [Function],
            }
          }
          data-testid="label"
        >
          Значение второго платежа
        </label>
        <div
          className={
            Object {
              "has": [Function],
              "is": [Function],
              "mix": [Function],
              "split": [Function],
              "state": [Function],
              "toString": [Function],
            }
          }
        >
          <div
            className={
              Object {
                "has": [Function],
                "is": [Function],
                "mix": [Function],
                "split": [Function],
                "state": [Function],
                "toString": [Function],
              }
            }
          >
            <div
              className="TextField TextField_size_s TextField_view_default TextField_form_default TextField_width_full TextField_type_text"
              data-testid="input"
              onKeyDown={[Function]}
            >
              <input
                autoFocus={false}
                className="TextField-Input"
                id="article_secondTimePaymentValue_2"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Значение"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className={
              Object {
                "has": [Function],
                "is": [Function],
                "mix": [Function],
                "split": [Function],
                "state": [Function],
                "toString": [Function],
              }
            }
          >
            <button
              className="Button Button_size_s Button_view_ghost Button_form_default Button_withIcon Button_onlyIcon MixFocus"
              onClick={[Function]}
              title="Опции"
              type="button"
            >
              <span
                className="Icon Icon_size_xs IconKebab Button-Icon"
              >
                <svg
                  className="Icon-Svg"
                  viewBox="0 0 16 16"
                >
                  <path
                    d="M8 4.5c.83 0 1.5-.67 1.5-1.5S8.83 1.5 8 1.5 6.5 2.17 6.5 3 7.17 4.5 8 4.5zm0 2c-.83 0-1.5.67-1.5 1.5S7.17 9.5 8 9.5 9.5 8.83 9.5 8 8.83 6.5 8 6.5zm0 5c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
